community: Chat Huggingface and HuggingFace TGI llms

minhna1112 opened the pull request:
community: 


  - **Description:** 
   - [x] The method `_resolve_model_id()` is only called when `model_id` is not explicitly defined, in order to avoid calling `list_inference_endpoints` when connecting `HuggingFaceTextGenInference` to a self-hosted TGI server.
  - [x]  For  `HuggingFaceTextGenInference` clients that send request to a HuggingFace-deployed Inference Endpoints, a Bearer Token is needed for authorization.
  - [x] More integrations tests is added for `HuggingFaceTextGenInference` 
  - **Issue:** Referring to this current [issue](https://github.com/huggingface/text-generation-inference/issues/747) in HuggingFace TGI repo
  - **Dependencies:** No dependencies for this change
  - **Twitter handle:**  @MinhN2804

Commit:
community: add API key into huggingface TGI validation

Commit:
community: Ignore `_resolve_model_id()` when calling a self-hosted HF TGI

Commit:
format and linting

Commit:
Add integration test for HuggingFace TGI

minhna1112 committed:
Merge branch 'master' into chat-huggingface

minhna1112 committed:
Merge branch 'master' into chat-huggingface

minhna1112 commented:
@hwchase17 Can you please help me merge this PR?

minhna1112 committed:
Merge branch 'master' into chat-huggingface

baskaryan reviewed (commented):
cc @andrewrreed

Commit:
Fix: Spelling error in

minhna1112 committed:
Merge branch 'master' into chat-huggingface

minhna1112 commented:
@andrewrreed  @baskaryan  I have fixed the spelling error occurred during the last check.

minhna1112 committed:
Merge branch 'master' into chat-huggingface

baskaryan commented on libs/community/langchain_community/llms/huggingface_text_gen_inference.py:
```diff
@@ -129,8 +131,17 @@ def build_extra(cls, values: Dict[str, Any]) -> Dict[str, Any]:
-        """Validate that python package exists in environment."""
-
+        """Validate that api key and python package exists in environment."""
+        huggingfacehub_api_token = get_from_dict_or_env(
+            values, "huggingfacehub_api_token", "HUGGINGFACEHUB_API_TOKEN"
+        )
```
will raise an error if no default is provided and token isn't provided, can we do
```suggestion
        huggingfacehub_api_token = get_from_dict_or_env(
            values, "huggingfacehub_api_token", "HUGGINGFACEHUB_API_TOKEN", default=""
        )
```

minhna1112 committed:
Merge branch 'master' into chat-huggingface

minhna1112 committed:
Update libs/community/langchain_community/llms/huggingface_text_gen_inference.py

Co-authored-by: Bagatur <22008038+baskaryan@users.noreply.github.com>

minhna1112 committed:
Merge branch 'master' into chat-huggingface

minhna1112 committed:
Merge branch 'master' into chat-huggingface

Commit:
Introduce `text_generation` dependency:
- Re-lock the `libs/community/poetry.lock` file
- Add `text-generation` to `libs/community/pyproject.toml`
- Add `pytest` for `libs/community/tests/integration_tests/llms/test_huggingface_text_gen_inference.py`
- Re-lint and format

minhna1112 commented:
@baskaryan  I have pushed [0f9784c](https://github.com/langchain-ai/langchain/pull/15582/commits/0f9784c94309953da4c88000f5a7493494b41b6e) to resolve library diffs during CI. Just tell me if there are more things need to be changed. Thanks a lot!

minhna1112 committed:
Merge branch 'langchain-ai:master' into chat-huggingface

minhna1112 committed:
Minor fix in `libs/community/pyproject.toml`

minhna1112 committed:
Merge branch 'chat-huggingface' of https://github.com/FSoft-AI4Code/langchain into chat-huggingface

minhna1112 committed:
Merge branch 'master' into chat-huggingface

minhna1112 committed:
Merge branch 'master' into chat-huggingface

minhna1112 commented:
@baskaryan  Can you please help me merge and close this PR? Thanks a lot!

minhna1112 committed:
Merge branch 'master' into chat-huggingface

minhna1112 committed:
Merge branch 'master' into chat-huggingface

minhna1112 committed:
Re-lock poetry

minhna1112 committed:
Merge branch 'master' into chat-huggingface

efriis commented:
closing because this is against the deprecated community huggingface integrations. Feel free to resubmit against the partner package (libs/partners/huggingface) if it's still missing there!
